pkg/synopsysctl/cmd_deploy.go

```diff
@@ -178,6 +184,123 @@ var deployCmd = &cobra.Command{
+		if isExist && err != nil {
+			return err
+		} else if isExist {
+			continue
+		}
+
+		switch crd {
+		case util.BlackDuckCRDName:
+			crdConfig = horizoncomponents.NewCustomResourceDefintion(horizonapi.CRDConfig{
+				APIVersion: "apiextensions.k8s.io/v1beta1",
+				Name:       util.BlackDuckCRDName,
+				Namespace:  namespace,
+				Group:      "synopsys.com",
+				CRDVersion: "v1",
+				Kind:       "Blackduck",
+				Plural:     "blackducks",
+				Singular:   "blackduck",
+				ShortNames: []string{"bds", "bd"},
+				Scope:      crdScope,
+			})
+		case util.AlertCRDName:
+			crdConfig = horizoncomponents.NewCustomResourceDefintion(horizonapi.CRDConfig{
+				APIVersion: "apiextensions.k8s.io/v1beta1",
+				Name:       util.AlertCRDName,
+				Namespace:  namespace,
+				Group:      "synopsys.com",
+				CRDVersion: "v1",
+				Kind:       "Alert",
+				Plural:     "alerts",
+				Singular:   "alert",
+				Scope:      crdScope,
+			})
+		case util.OpsSightCRDName:
+			crdConfig = horizoncomponents.NewCustomResourceDefintion(horizonapi.CRDConfig{
+				APIVersion: "apiextensions.k8s.io/v1beta1",
+				Name:       util.OpsSightCRDName,
+				Namespace:  namespace,
+				Group:      "synopsys.com",
+				CRDVersion: "v1",
+				Kind:       "OpsSight",
+				Plural:     "opssights",
+				Singular:   "opssight",
+				ShortNames: []string{"ops"},
+				Scope:      crdScope,
+			})
+		}
+		if crdConfig != nil {
+			crdConfig.AddLabels(map[string]string{"app": "synopsys-operator", "component": "operator", fmt.Sprintf("synopsys.com/operator.%s", namespace): namespace})
+			deployer.Deployer.AddComponent(horizonapi.CRDComponent, crdConfig)
+			isAdded = true
+		}
+	}
+
+	if isAdded {
+		err := deployer.Deployer.Run()
+		if err != nil {
+			return errors.Annotate(err, "unable to deploy custom resource definition")
+		}
+	}
+
+	return nil
+}
+
+func checkAndUpdateCustomResource(crdType string, namespace string, scope apiextensions.ResourceScope, isClusterScoped bool) (bool, error) {
+	crd, err := util.GetCustomResourceDefinition(apiExtensionClient, crdType)
+	if err != nil {
+		return false, fmt.Errorf("unable to get %s custom resource definition due to %+v", crdType, err)
+	}
+
+	// CRD exist with different scope and hence error out
+	if scope != crd.Spec.Scope {
+		return true, fmt.Errorf("already %s custom resource definition exist with %v scope. updating the %s custom resource definition scope is not supported. please contact the Synopsys support team to handle it", crd.Name, crd.Spec.Scope, crd.Name)
+	}
+
+	if isClusterScoped {
+		for key, value := range crd.Labels {
+			if strings.HasPrefix(key, "synopsys.com/operator.") {
+				if value != namespace {
+					return true, fmt.Errorf("already Synopsys Operator is watching for %s custom resource events in namespace '%s'. Only one Synopsys Operator is allowed to watch %s custom resource events. Please contact Synopsys support in case of any queries", crdType, value, crdType)
```

Comment:
there is already a Synopsys Operator managing {crdType} in namespace {ns}.  Only one Synopsys Operator may manage {crdType} per namespace

^^ Does that sound about right?

Reply:
done